core/http-auth-aws/src/main/java/software/amazon/awssdk/http/auth/aws/internal/signer/chunkedencoding/ChunkedEncodedPublisher.java

```diff
@@ -138,24 +143,46 @@ private ByteBuffer encodeChunk(ByteBuffer byteBuffer) {
 
         int extensionsLength = calculateExtensionsLength(chunkExtensions);
 
-        int encodedLen = chunkSizeHex.length + extensionsLength + CRLF.length + contentLen + CRLF.length;
+        boolean isTrailerChunk = contentLen == 0;
+
+        List<ByteBuffer> trailerData;
+        if (isTrailerChunk) {
```

Comment:
what happens if when we are uploading a empty/touch file , in this case the contetnt length will be Zero but there will trailer checksums for empty files.